### ISSUE TITLE
Refactor: Draw should be separate from geometry computation.

### ISSUE THREAD

**Issue by jdtoy** [geometrygeeks/visual-comparison#5]
Currently both steps [1A and 2](https://github.com/geometrygeeks/visual-comparison/wiki/Drawing-a-Bike) are in the `parse-geometry.js` module.

It would be better to split these into two.

https://github.com/geometrygeeks/visual-comparison/blob/149aef365f7a034a7e0a6882e44bebe8d1510da3/src/parse-geometry.js#L152-L254

**Comment by moarrubber** [comment]
Could you clarify what you mean by split these in two? Do you mean have a step where the geometry data is parsed, calculated, and formed into a standard set of definitions for the draw step to be able to print in a dumb way? If so that sounds terrific (and makes a good separation for unit testing).

What do you think the hand-off between the two steps looks like?

**Comment by jdtoy** [comment]
Yes. Exactly. For those reasons, and also because it will ease sharing development.

Ideally I'd like to resolve the geometry parameters to a set of points ([wiki:Points on a Bike](https://github.com/geometrygeeks/visual-comparison/wiki/Points-on-a-Bike)), then have a separate method draw those points as a bike, styling as appropriate for Road/MTB/Tri.

Later others may wish to reuse the geometry parsing code, but switch to a different visualisation library (i.e. replace Snap.svg).

Hand off is unclear at the moment. Suggestions very welcome.
- bike.draw() or Draw(bike) ?
- separation of all drawing implementation details into the 'drawing' module.
    - there's some Snap.svg related details lines 77-90 of visual-bike.js

https://github.com/geometrygeeks/visual-comparison/blob/149aef365f7a034a7e0a6882e44bebe8d1510da3/src/visual-bike.js#L77-L90

**Comment by jdtoy** [comment]
FYI: I've done some initial playing around, but don't think this is suitable yet.
https://github.com/jdtoy/visual-comparison/tree/feature/refactor-draw/src

**Comment by jdtoy** [comment]
@moarrubber I've refactored as discussed. Any closing thoughts?

**Comment by moarrubber** [comment]
@jdtoy that is a huge improvement./

What are your thoughts on the difference between "this.resolvedPoint.seat_tube_top_x" vs "this.resolvedPoint.seat_tube_top.x"?
1. The point for the seat_tube_top is logically a "thing" with properties x and y. Or would seat_tube make sense as an object containing two points for the endpoints?
2. I think it would make navigating the code a little easier, being able to double click to highlight and easily find the other places where a particular point is modified.
I think the counterargument is that the usability difference is minor, and additional levels to the container heirarchy could obfuscate the code.

There is a lot of repetition in the attributes passed to drawObjs. Sometimes fill, stroke, and strokeWidth are provided, sometimes only strokeWidth. Can these be defined once and re-used? I can discern at least 3 different categories of attributes: wheels, frame, and virtual features.

**Comment by jdtoy** [comment]
@moarrubber Good to hear. 

> What are your thoughts on the difference between "this.resolvedPoint.seat_tube_top_x" vs "this.resolvedPoint.seat_tube_top.x"?

Yes to points being Point objects - I've opened https://github.com/geometrygeeks/visual-comparison/issues/13 for this. At this stage I think it might be better to wait and see on creating {seat|top|head|...} Tube objects, as I can't yet imagine a use case for interacting with a Tube object, whereas the use case for working with Points is very clear.

> There is a lot of repetition in the attributes passed to drawObjs. Sometimes fill, stroke, and strokeWidth are provided, sometimes only strokeWidth. Can these be defined once and re-used? I can discern at least 3 different categories of attributes: wheels, frame, and virtual features.

I agree. I've opened https://github.com/geometrygeeks/visual-comparison/issues/14.
In addition I'm aware that the library will need to have slightly different Draw()s for road/mtb/tri/tt, and default drawing parameters are likely to come into that too.
